util/mkconf.c

```diff
@@ -0,0 +1,142 @@
+  { "bsd-api",          "BSD_API",          "Include BSD-sockets" },
+  { "bsd-fatal",        "BSD_FATAL",        "Exit application on BSD-socket fatal errors" },
+  { "bootp",            "BOOTP",            "Include BOOTP client code" },
+  { "dhcp",             "DHCP",             "Include DHCP boot client code" },
+  { "rarp",             "RARP",             "Include RARP boot client code.  Contributed by Dan Kegel." },
+  { "geoip",            "GEOIP",            "Include GeoIP support. From Tor's geoip.c" },
+  { "ipv6",             "IPV6",             "Include IPv6 dual-stack support" },
+  { "language",         "LANGUAGE",         "Include Language translation code" },
+  { "fragments",        "FRAGMENTS",        "Include IP-fragment handling" },
+  { "statistics",       "STATISTICS",       "Include protocol statistics count" },
+  { "stackwalker",      "STACKWALKER",      "Use StackWalker for allocations and crash-dumps." },
+  { "fsext",            "FSEXT",            "Use djgpp's File Extensions for file I/O functions" },
+  { "loopback",         "LOOPBACK",         "Use the simple loopback device" },
+  { "embedded",         "EMBEDDED",         "Make an embeddable (ROM-able) target.  See note in config.h" },
+  { "buffered-io",      "BUFFERED_IO",      "Use buffered file I/O in pcconfig.c.  Otherwise use a simple read-ahead cache" },
+  { "tftp",             "TFTP",             "Include TFTP protocol for simple file retrival" },
+  { "udp-only",         "UDP_ONLY",         "Exclude all stuff related to the TCP protocol" },
+  { "tcp-sack",         "TCP_SACK",         "Include TCP Selective ACK feature (not yet)" },
+  { "echo-disc",        "ECHO_DISC",        "Include echo/discard servers" },
+  { "pppoe",            "PPPOE",            "PPP-over-Ethernet encapsulation" },
+  { "fast-pkt",         "FAST_PKT",         "Use faster all real-mode PKTDRVR receiver (DOSX only)" },
+  { "dead-gwd",         "DEAD_GWD",         "Dead Gateway detection in ARP module" },
+  { "gzip",             "GZIP",             "Use gzip compression for PPP and/or pcap debug file.  Also for SCTP." },
+  { "profiler",         "PROFILER",         "Include simple execution profiler (only DOSX with 64-bit types)" },
+  { "tcp-md5",          "TCP_MD5",          "TCP MD5 signature option" },
+  { "idna",             "IDNA",             "Support national characters in DNS name lookups." },
+  { "dynip-cli",        "DYNIP_CLI",        "Support DynDNS.org name/IP client" },
+  { "sctp",             "SCTP",             "Stream Control Transfer Protocol; experimental" }
+};
+
+#define NUM_OPTIONS (sizeof (options) / sizeof (struct cfgopt))
+
+static enum optval option_values[NUM_OPTIONS] = { };
+
+static int skip (const char* prefix, const char** arg)
+{
+  const size_t len = strlen (prefix);
+  if (strncmp (prefix, *arg, len) == 0)
+  {
+    *arg += len;
+    return (1);
+  }
+  return (0);
+}
+
+static void usage ()
```

Comment:
clang-cl may not like such old K&R style. Better use `usage (void)`.

Reply:
Oh right, there's a difference between those.  Must say I'm more familiar with C++ than C.  Will fix.